recipe_scrapers/__init__.py

```diff
@@ -605,22 +606,40 @@ def scraper_exists_for(url_path: str) -> bool:
     return host_name in get_supported_urls()
 
 
+def check_scraper_allowed(url_path, host_name, **options: Any) -> bool:
+    """Checks the website's robots.txt file to see if scraping is allowed.
+    If scraping is allowed, returns true. Note that this does not check to see
+    when the last time robots.txt was checked, so this could lead to constant pings
+    on robots.txt"""
+    if not options.get("check_robots_txt", True):
+        return True
+    else:
+        url = urljoin(url_path, '/robots.txt')
+        resp = requests.get(url, headers=HEADERS)
+        rp = urllib.robotparser.RobotFileParser()
+        rp.parse(resp.text)
+        if rp.can_fetch('*', url_path):
+            return True
+        else:
+            raise AutomatedClientsNotAllowed(host_name)
+
+
 def scrape_me(url_path: str, **options: Any) -> AbstractScraper:
     host_name = get_host_name(url_path)
+    if check_scraper_allowed(url_path, host_name, **options):
```

Comment:
I'd use the `assert <method_name>(...)` here and avoid the nesting that comes with the if. Reads better to me. You may ignore my comment here if you like the if syntax more.